### DISCUSSION TITLE
Question about crack installer code script

### DISCUSSION THREAD

**Discussion by evolv30101** [Dilbert66/esphome-dsckeybus#143] [General]
Please forgive my ignorance....im not a programmer by any means.  

I have an alarm i recently integrated in HA using your repo which i believe is working good, i see sensors etc....and they change when triggered (ie..door opened).  So i believe that is good, lol.  The issue im currently stuck on is the alarm has at least one if not 2 bad sensors that need replaced which i believe are wireless based on the devices i found.  I have the master but I do not have the installer code for the alarm.  So im currently trying to run the crack script to find it. 

I managed to get python and the script running (api version) but im not sure its actually working correctly.  I replaced the variables in the script to what i believe was wanted and the script started to run.  

Right now its just printing:  "sending ##*8" then a few secs later "no response".  Not sure what the expected output looks like, i tried to decipher some of the code but again im not a programmer.  I was sort of expecting something like this:  "sending code:",test_code"  to output on the screen.  As a side note i also have 03:zones open currently due to the bad sensors if it matters.  

Does this sound correct or like its not working?  Any help much appreciated!

**Comment by evolv30101** [comment]
Just an update....I did try the web version of the script as well but got the same result.  

Tinkering around i was able to scroll thru the open zones in using the keypad on the web server using > and it shows the open zones and allows me to go thru so i believe it is communicating.  Again ... i have no idea what im doing, even more so with alarm systems apparently :)

**Comment by Dilbert66** [comment]
I would need to see the esphome logs as the crack script is running to see what is happening. If you entered the right address and key it should work.

**Comment by evolv30101** [comment]
Sorry for delay, this is at remote location and the workstation locked me out for a little bit.  

Here's two logs, one with the script trying to run and then another with me pressing the ">" scrolling thru  the open zones.  

[Alarm_With_Script_running.txt](https://github.com/user-attachments/files/17462047/Alarm_With_Script_running.txt)

[Alarm_With_Keypresses.txt](https://github.com/user-attachments/files/17462055/Alarm_With_Keypresses.txt)

**Comment by Dilbert66** [comment]
That is very odd.  With the script running, it's definitively receiving the cmds so your api setup looks ok but since the panel is not responding at all, it's as if the esp to panel send circuit is not working but yet, on your keypressesit shows it's communicating.   Can you try typing "##*8" on the web keypad itself and capture the esplog just to confirm that it's working as it should.  
This is an example of what you should see when sending ##*8
```
I][dscalarm:767]: Writing keys: ##*8 to partition 1, partition disabled: 0 , partition locked: 0
[I][dscalarm:1985]: status 8F, last status 8F,selection 01,partition=1,skip=0,force=1
[I][Moduledata:922]: 05: FF 01 2D FF FF FF FF FF FF FF 01 00 00 00 00 00  (74)
[I][Paneldata:922]: 5D: 5D 00 00 00 00 00 00 5D 00 00 00 00 00 00 00 00  (58)
[I][Paneldata:922]: E6: E6 00 18 01 00 00 00 00 00 FF 00 00 00 00 00 00  (74)
[I][Paneldata:922]: 05: 05 00 11 8F 11 01 11 03 11 C7 01 00 00 00 00 00  (74)
[I][Moduledata:922]: 05: FF 01 2D FF FF FF FF FF FF FF 01 00 00 00 00 00  (74)
[I][Moduledata:922]: 05: FF 01 28 FF FF FF FF FF FF FF 01 00 00 00 00 00  (74)
[I][Paneldata:922]: 5D: 5D 00 00 00 00 00 00 5D 00 00 00 00 00 00 00 00  (58)
[I][Paneldata:922]: E6: E6 00 18 01 00 00 00 00 00 FF 00 00 00 00 00 00  (74)
```

**Comment by evolv30101** [comment]
Odd indeed....i built two of these, one for each building at a church and the other one works fine (well other than i got stuck trying to troubleshoot all the pre-existing issues i found with their alarms afterwards, lol).  Same buildout\parts.  My esp build for both alarms was a Wesp32 using POE for power and the "isolated" circuit design shown on your page.  

I have power cycled the alarm in question just in case as it seemed sort of "glitchy".  Didn't really do anything to help.  I'll just throw this in here as a "just in case" since I'm not an electrical engineer either... this pc1864 in particular has two keypads on it plus my esp.  Not sure if some how I'm drawing too much power w/ multiple keypads or something stupid.   I wouldn't think so since im using the wesp32's poe for power.  

There are definitely issues with this system in question as well, i have 2 sensors which i believe are wireless that need replaced.  Doors in question are always open or sporadically open when shouldn't be.  I've tried replacing batteries but didn't help.  Looks like they tapped in to the existing hard wire to the doors and then used a wireless sensor to communicate with one of the keypads which i believe has wireless capability.  Still trying to map everything out as the info they had on hand was ... lets just say ... nowhere near accurate or up to date.  Again ... I'm not an alarm guy and not sure if any of this could be playing into it, just throwing it up in case something stands out to you.

I ordered some more small bread boards which should arrive shortly.  Will probably just rebuild that circuit in question just in case i have a flakey transistor or opto in there.  Kind of at a loss.

First presses are using JS panel.   Second press are using the esp's web panel.  I think i accidently hit an extra 8 in there...:)

[Alarm_Star8.txt](https://github.com/user-attachments/files/17522969/Alarm_Star8.txt)

**Comment by Dilbert66** [comment]
Ok, had a look at your log. The panel is definitively not responding to the key presses.  For instance, when you enter * it should switch the display to "press \<*\> for zone bypass" or something like it.  You should see a response.  The "#" key is just to clear it out of any menu option it might be in and get it back to the normal display.   The "*8" keys are to get into the installer menu. 
So, it still points to an issue with the TX circuit.

**Reply by evolv30101** [reply]
OK.  I'll proceed and try to rebuild the circuit with fresh parts and see if i get a different result.

The only keys that seems to respond is the < or > keys.  Unless its just an artifact with the software interface since i have open zones and its already memorized the zones leading me to believe they are responding when in reality they aren't.  

I think my breadboards arrive tomorrow so i'll whip up another and thru it in on Tuesday most likely when im on site again.

**Comment by brucehat** [comment]
I have 4 keypads with auxiliary power no issues on 1864 if it helps. Moved my install from 1832 at previous house. Cheers BruceOn 25 Oct 2024, at 19:42, Alain Turbide ***@***.***> wrote:﻿
Ok, had a look at your log. The panel is definitively not responding to the key presses.  For instance, when you enter "" it should switch the display to "press <> for zone bypass" or something like it.  You should see a response.  The "#" key is just to clear it out of any menu option it might be in and get it back to the normal display.   The "*8" keys are to get into the installer menu.
So, it still points to an issue with the TX circuit.

**Reply by evolv30101** [reply]
Good to know.  I didn't think i could be over drawing but i don't know these alarm systems at all.

**Comment by Dilbert66** [comment]
Also double check the pins used in the yaml.  You can try out the resistor divider circuit also as I find it's the least problematic.

**Reply by evolv30101** [reply]
I'll double check the wiring again on the esp pins.  yaml config is good, i copied the same config to other alarm and just changed names/sensors/etc.  to keep everything consistent with a template.  

I dont have any pn2222a's.  I would need to pick some up somewhere.   I believe i have some ZVNL120A laying around from a different project.  Is that a suitable replacement?

**Comment by Dilbert66** [comment]
Any small signal npn transistor will work fine.   That zvnl120a looks to be a mosfet?  I suppose that can work but I've never tried it

**Comment by evolv30101** [comment]
Just an update here.....rebuilt new circuit and installed today = success!  

Not sure what it was...possibly bad transistor/resistor....whatever.   

Crack running for few mins now and already thru about 600 codes.

**Comment by Dilbert66** [comment]
Glad to hear. Let me know if you get your code.  I've seen the very odd occasion that it might miss the code but you can just run it again.

**Comment by evolv30101** [comment]
WIll do.

No chance there is a zone configuration discovery script floating around somewhere that will loop thru and output existing configuration for the zones on the system is there?    IE zone, type, attributes, serial# (wireless) to csv.   Or even existing user codes/slots?   

I know....im asking a lot :)

**Comment by Dilbert66** [comment]
Not that I know of... I suppose I could write one some day. Just don't have the time right now unfortunately.  You can easily map that yourself by going through the programming once you get the code. If you don't have an lcd keypad, you can use the web one.

**Reply by evolv30101** [reply]
It made me sweat for it....but it eventually found the code in the low 9000's :)

This is my was my game plan once i got the code.  Do it from the console so i can look up alarm commands etc and copy to excel or something.  I think one building is pretty accurate (~85-90%) as far as what i was able to dig up but the other building is an older system  age wise and put in by a different company and is a mess.  I think the servicing company just kept bandaid'ing and it became a rats nest.

**Comment by Dilbert66** [comment]
Just get some online programming worksheets for your panel version and go through the programs and fill out the sheets as go. It goes pretty fast

**Reply by evolv30101** [reply]
Had to google that..... isnt that nifty.
